src/contexts/Photos/Photos.context.tsx

```diff
@@ -198,33 +207,26 @@ export const PhotosContextProvider: React.FC = ({ children }) => {
     }
 
     selection.resetSelectionMode();
-
-    if (syncStatus === PhotosSyncStatus.Completed) {
-      setSyncStatus(PhotosSyncStatus.Unknown);
-      setTotalTasks(0);
-      setPendingTasks(0);
-      setFailedTasks(0);
-      setCompletedTasks(0);
-    }
-
-    setUploadedPhotosItems([]);
+    accumulatedDevicePhotosItems = [];
+    setSyncStatus(PhotosSyncStatus.Unknown);
+    setTotalTasks(0);
+    setPendingTasks(0);
+    setFailedTasks(0);
+    setCompletedTasks(0);
+    setPhotosInLocalDB(0);
+    setPhotosInRemote(0);
+    setPhotosInDevice(0);
   }
-  async function onRemotePhotosSynced(photosItem: PhotosItem) {
-    photosLogger.info('Checking db for new photos');
-
-    const existsSynced = syncedPhotosItems.current.find(
-      (syncedPhotosItem) =>
-        syncedPhotosItem.name === photosItem.name && syncedPhotosItem.takenAt === photosItem.takenAt,
-    );
-    if (existsSynced) return;
-    syncedPhotosItems.current.push(photosItem);
 
+  async function onRemotePhotosSynced() {
+    const syncedPhotos = await photos.realm.getSyncedPhotos();
+    syncedPhotosItems.current = syncedPhotos.map((syncedPhoto) => photos.utils.getPhotosItem(syncedPhoto));
     setPhotosInLocalDB(syncedPhotosItems.current.length);
     const mergedPhotosItems = photosUtils.mergePhotosItems(devicePhotosItems.current.concat(syncedPhotosItems.current));
     setDataSource(
       new DataProvider(function (r1, r2) {
```

Comment:
what is `r1`, what is `r2`?

Reply:
row1, row2, bad naming I aggree.

This function compares "what is different" to detect if we should render again or not